### PR TITLE
#Implement redux store

### PR DESCRIPTION
**Pull request by phemonick**
### What does this PR do
- [x] Implement redux for state management

#### Description of Task to be completed?
- [x] Add redux dependencies
- [x] connect redux store to base app

#### How should this be manually tested?
- Clone this Repo
- `expect app to work properly`

#### Any background context you want to provide?
- No

### CODE REVIEW

File: src/redux/reducers/userReducer.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { USER_INFO } from '../actions/types';
+
+const userReducer = (state = { users: { data: { users: [] } } }, action) => {
```
Comment (kingisaac95) [inline src/redux/reducers/userReducer.js]: Users data is deeply nested. Might wanna fix that from the backend

File: src/redux/reducers/index.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { combineReducers } from 'redux';
+import users from './userReducer';
+
+const root = combineReducers({
```
Comment (kingisaac95) [inline src/redux/reducers/index.js]: This should be a one-liner

File: src/redux/actions/index.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+});
+
+export const notAuthenticated = () => ({
+  type: IS_LOGGED_IN,
+  payload: false
+});
+
+export const isAuthenticated = () => dispatch => {
+  const jwtToken = window.localStorage.getItem('userToken');
+  if (!jwtToken || !jwtToken.length > 9) {
```
Comment (kingisaac95) [inline src/redux/actions/index.js]: Why are you doing a check of token length?

File: src/redux/actions/index.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    dispatch(notAuthenticated());
+  } else {
+    // check if token has expired
+    const decoded = jwtDecode(jwtToken);
+    const { email } = decoded;
+    const name = email.split('.');
```
Comment (kingisaac95) [inline src/redux/actions/index.js]: What if the email is not my name? Eg. `coolguy@gmail.com`

File: src/redux/actions/index.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+});
+
+export const isAuthenticated = () => dispatch => {
+  const jwtToken = window.localStorage.getItem('userToken');
+  if (!jwtToken || !jwtToken.length > 9) {
+    dispatch(notAuthenticated());
+  } else {
+    // check if token has expired
+    const decoded = jwtDecode(jwtToken);
+    const { email } = decoded;
```
Comment (kingisaac95) [inline src/redux/actions/index.js]: Don't store emails in local storage. Only put none sensitive data in there

File: src/redux/actions/index.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import jwtDecode from 'jwt-decode';
+import { IS_LOGGED_IN } from './types';
+
+export const authenticated = (name) => ({
+  type: IS_LOGGED_IN,
+  payload: true,
```
Comment (kingisaac95) [inline src/redux/actions/index.js]: Why do you have this `payload` key?

File: package.json [hunk 7]
```diff
@@ -3,9 +3,19 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "axios": "^0.18.0",
```
Comment (kingisaac95) [inline package.json]: Lock package versions by removing `^`

File: .env [hunk 8]
```diff
@@ -0,0 +1 @@
+SKIP_PREFLIGHT_CHECK=true
```
Comment (kingisaac95) [inline .env]: You should not have your env file on GitHub. It should be ignored by git

File: src/redux/actions/index.js (hunk 5 above)
Comment (phemonick) [reply src/redux/actions/index.js]: Ok, It will be replaced with an ID.

**Pull request merged**